coordinator/schedulepod/main.go

```diff
@@ -0,0 +1,694 @@
+		return err
+	}
+	sourceComplete, err := client.GetSourceVariant(context.Background(), metadata.NameVariant{Name: sourceName, Variant: ""})
+	if err != nil {
+		return fmt.Errorf("could not get source variant")
+	}
+	if metadata.READY != sourceComplete.Status() {
+		return fmt.Errorf("source variant not set to ready once job completes")
+	}
+	transformationQuery := fmt.Sprintf("SELECT * FROM {{%s.}}", sourceName)
+	transformationName := strings.Replace(createSafeUUID(), "-", "", -1)
+	transformationID := metadata.ResourceID{Name: transformationName, Variant: "", Type: metadata.SOURCE_VARIANT}
+	sourceNameVariants := []metadata.NameVariant{{Name: sourceName, Variant: ""}}
+	if err := createTransformationWithProvider(client, serialPGConfig, transformationName, transformationQuery, sourceNameVariants, "*/1 * * * *"); err != nil {
+		return err
+	}
+	transformationCreated, err := client.GetSourceVariant(context.Background(), metadata.NameVariant{Name: transformationName, Variant: ""})
+	if err != nil {
+		return fmt.Errorf("could not get transformation: %v", err)
+	}
+	if transformationCreated.Status() != metadata.CREATED {
+		return fmt.Errorf("Transformation not set to created with no coordinator running")
+	}
+	go func() {
+		if err := coord.WatchForNewJobs(); err != nil {
+			logger.Errorf("Error watching for new jobs: %v", err)
+		}
+	}()
+	go func() {
+		if err := coord.WatchForUpdateEvents(); err != nil {
+			logger.Errorf("Error watching for new update events: %v", err)
+		}
+	}()
+	time.Sleep(70 * time.Second)
+	jobClient, err := runner.NewKubernetesJobClient(runner.GetCronJobName(transformationID), runner.Namespace)
+	if err != nil {
+		return err
+	}
+	cronJob, err := jobClient.GetCronJob()
+	if err != nil {
+		return err
+	}
+	lastExecutionTime := cronJob.Status.LastSuccessfulTime
+	if lastExecutionTime.IsZero() {
+		return fmt.Errorf("job did not execute in time")
+	}
+	// transformationUpdated, err := client.GetSourceVariant(context.Background(), metadata.NameVariant{Name: transformationID.Name, Variant: ""})
+	// if err != nil {
+	// 	return err
+	// }
+	// transformationUpdateStatus := transformationUpdated.UpdateStatus()
+	// transformationLastUpdated := transformationUpdateStatus.LastUpdated
+	// if transformationLastUpdated.AsTime().IsZero() {
+	// 	return fmt.Errorf("Scheduler did not update transformation")
+	// }
+	return nil
+}
+
+func CreateOriginalPostgresTable(tableName string) error {
+	url := fmt.Sprintf("postgres://%s:%s@%s:%s/%s", postgresConfig.Username, postgresConfig.Password, postgresConfig.Host, postgresConfig.Port, postgresConfig.Database)
+	ctx := context.Background()
+	conn, err := pgxpool.Connect(ctx, url)
+	if err != nil {
+		return err
+	}
+	createTableQuery := fmt.Sprintf("CREATE TABLE %s (entity VARCHAR, value INT, ts TIMESTAMPTZ)", sanitize(tableName))
+	if _, err := conn.Exec(context.Background(), createTableQuery); err != nil {
+		return err
+	}
+	for _, record := range testOfflineTableValues {
```

Comment:
Do you even need records for this?

Reply:
yes records confirm the underlying resource changes after update